### PR TITLE
Remove start dates from Checkout

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Remove the start date inputs from the checkout.

Move the hook to update start date to the `PurchaseForm` scope.

## Justify why they are needed

Design changes.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

[GRW-1817]: https://hedvig.atlassian.net/browse/GRW-1817?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/store/src/components/ProductPage/PurchaseForm/useUpdateStartDate.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  return [
+    handleUpdateStartDate,
+    {
+      loading: result.loading,
+      userError,
+    },
+  ] as const
+}
+
+const convertToDate = (value: unknown) => {
```
Comment (alebedev) [inline apps/store/src/components/ProductPage/PurchaseForm/useUpdateStartDate.ts:58]: Is it time to create utils/date?

**Review by alebedev — APPROVED** [review]
Looks good, one small comment inside

File: apps/store/src/components/ProductPage/PurchaseForm/useUpdateStartDate.ts (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/components/ProductPage/PurchaseForm/useUpdateStartDate.ts:58]: Perhaps, do we have anywhere else we need this?

Comment (alebedev) [reply apps/store/src/components/ProductPage/PurchaseForm/useUpdateStartDate.ts:58]: Already did it locally, so let's consider this resolved

**Pull request merged**